lib/ios-log.js

```diff
@@ -0,0 +1,217 @@
+      if (logsPath.indexOf('*') >= 0) {
+        return; 
+      }
+        await mkdirp(logsPath);
+        await touch(path.resolve(logsPath, 'system.log'));
+        await fs.appendFile(path.resolve(logsPath, 'system.log'), 
+              'A new Simulator is about to start!'); 
+        let files;
+        try {
+          files = await glob(path.resolve(logsPath, 'system.log'));
+        if (files.length < 1) {
+          throw 'Could not start log capture';
+        }
+      } catch (e) {
+        logger.error('Could not start log capture because no iOS ' +
+        'simulator logs could be found at ' + logsPath + '/system.log' +
+        'Logging will not be functional for this run');
+      }
+
+      let lastModifiedLogPath = files[0];
+      let lastModifiedLogTime = await fs.stat(files[0]).mtime; //promisify this 
+      for (let file in files) {
+        let mtime = await fs.stat(file).mtime;
+        if (mtime > lastModifiedLogTime) {
+          lastModifiedLogPath = file;
+          lastModifiedLogTime = mtime;
+        }
+      }
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+      this.finishStartingLogCapture();
+    } catch (err) {
+      logger.warn('System log capture failed.');
+      return;
+    }    
+  }  
+
+  //set up the procedure with stdout stderr, once it has any output -- > resolve the promise, show logs have started
+  // 
+
+  // first time you want to ignore it, 
+  // have the promise use a local varialbe outside of promise, and change the boolean after the first line is done. 
+  // make first lien empty 
+  finishStartingLogCapture () {
+    if (!this.proc) {
+      let msg = 'Could not capture device log';
+      logger.warn(msg);
+      throw new Error(msg);
+    }
+     this.startTimer = setTimeout(function () {
+      let msg = 'Log capture did not start in a reasonable amount of time';
+      logger.error(msg);
+
+      throw new Error(msg);
+      
+    }, START_TIMEOUT);
+    let firstLine = true;
+    return new Promise((resolve, reject) => {
+      this.proc.on('output', (stdout, stderr) => {
+        if (stdout) {
+          //log.info(`[STDOUT] ${stdout.trim()}`);
+          clearTimeout(this.startTimer);
+          if (firstLine) {
+            this.onOutput(stdout);
+          } else {
+            this.logRow += stdout;
+            if (stdout.substr(-1,1) === '\n') {
+              this.onOutput(stdout, '');
+              this.logRow = ''; 
+              firstLine = false;
+            }
+          }         
+          resolve();
+        }
+        if (stderr) {
+          //log.info(`[STDERR] ${stderr.trim()}`);
+          clearTimeout(this.startTimer);
+          if(/execvp\(\)/.test(stderr)) {
+            logger.error('iOS log capture process failed to start');
+            reject(new Error('iOS log capture process failed to start'));
+            return;
+            }
+          reject(stderr);
+          this.onOutput(stderr, ' STDERR');
```

Comment:
I am not sure if this will print after rejection? @jlipps what do you think, in theory it should print but i am not sure